test/kad-dht.spec.js

```diff
@@ -222,6 +222,39 @@ describe('KadDHT', () => {
     })
   })
 
+  it('put - get with update', function (done) {
+    this.timeout(10 * 1000)
+    const tdht = new TestDHT()
+
+    tdht.spawn(3, (err, dhts) => {
+      expect(err).to.not.exist()
+      const dhtA = dhts[0]
+      const dhtB = dhts[1]
+      const dhtC = dhts[2]
+
+      parallel([
+        (cb) => connect(dhtA, dhtB, cb),
+        (cb) => connect(dhtA, dhtC, cb),
+        (cb) => dhtA.put(Buffer.from('/v/hello'), Buffer.from('worldA'), cb),
+        (cb) => dhtB.put(Buffer.from('/v/hello'), Buffer.from('worldB'), cb),
+        (cb) => dhtC.put(Buffer.from('/v/hello'), Buffer.from('worldC'), cb)
+      ], (err) => {
+        expect(err).to.not.exist()
+        parallel([
+          (cb) => dhtA.get(Buffer.from('/v/hello'), { maxTimeout: 1000 }, cb),
+          (cb) => dhtB.get(Buffer.from('/v/hello'), { maxTimeout: 1000 }, cb),
+          (cb) => dhtC.get(Buffer.from('/v/hello'), { maxTimeout: 1000 }, cb)
+        ], (err, res) => {
+          expect(err).to.not.exist()
+          expect(res[0]).to.eql(Buffer.from('worldA'))
+          expect(res[1]).to.eql(Buffer.from('worldB'))
+          expect(res[2]).to.eql(Buffer.from('worldC'))
```

Comment:
So looking at this test, I'm curious what actually happens with `_putValueToPeer`. If each dht ends up getting it's own record, are they going to tell their peers to update their records? Which record wins in this scenario, or do they all keep their own?

Reply:
Because they are still resyncing. If I do the following:

```js
    parallel([
        (cb) => connect(dhtA, dhtB, cb),
        (cb) => connect(dhtA, dhtC, cb),
        (cb) => dhtA.put(Buffer.from('/v/hello'), Buffer.from('worldA'), cb),
        (cb) => dhtB.put(Buffer.from('/v/hello'), Buffer.from('worldB'), cb),
        (cb) => dhtC.put(Buffer.from('/v/hello'), Buffer.from('worldC'), cb)
      ], (err) => {
        expect(err).to.not.exist()
        parallel([
          (cb) => dhtA.get(Buffer.from('/v/hello'), { maxTimeout: 1000 }, cb),
          (cb) => dhtB.get(Buffer.from('/v/hello'), { maxTimeout: 1000 }, cb),
          (cb) => dhtC.get(Buffer.from('/v/hello'), { maxTimeout: 1000 }, cb)
        ], (err, res) => {
          expect(err).to.not.exist()
          expect(res[0]).to.eql(Buffer.from('worldA'))
          expect(res[1]).to.eql(Buffer.from('worldB'))
          expect(res[2]).to.eql(Buffer.from('worldC'))

          setTimeout(() => {

            parallel([
              (cb) => dhtA.get(Buffer.from('/v/hello'), { maxTimeout: 1000 }, cb),
              (cb) => dhtB.get(Buffer.from('/v/hello'), { maxTimeout: 1000 }, cb),
              (cb) => dhtC.get(Buffer.from('/v/hello'), { maxTimeout: 1000 }, cb)
            ], (err, res) => {
              expect(err).to.not.exist()
              console.log('res', res)
              expect(res[0]).to.eql(Buffer.from('worldA'))
              expect(res[1]).to.eql(Buffer.from('worldB'))
              expect(res[2]).to.eql(Buffer.from('worldC'))

              tdht.teardown(done)
            })
          }, 2000)
```

This will fail, as they will be hopefully synced by that time. In the `_putValueToPeer`, it will send new messages to the remaing peers. Then, they will use the creationTime to know about the best record.